druid-shell/src/platform/windows/window.rs

```diff
@@ -1083,28 +1044,17 @@ impl WndProc for MyWndProc {
 
                 Some(0)
             }
-            WM_CLOSE => {
-                if let Ok(mut s) = self.state.try_borrow_mut() {
-                    let s = s.as_mut().unwrap();
-                    s.handler.request_close();
-                    Some(0)
-                } else {
-                    None
-                }
-            }
+            WM_CLOSE => self
+                .with_wnd_state(|s| s.handler.request_close())
+                .map(|_| 0),
             DS_REQUEST_DESTROY => {
                 unsafe {
                     DestroyWindow(hwnd);
                 }
                 Some(0)
             }
```

Comment:
If we make WindowHandle::close() deferred I think we could get rid of this custom message now,
unless there is some other reason we need it?